File: 2024-2/0682/unidad06.d_Web_04/SoftPub/SoftPubPersonaBussiness/EspecialidadBo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPubPersonaControler.DAO;
using SoftPubPersonaControler.DAOImpl;
using SoftPubPersonaModel;

namespace SoftPubPersonaBussiness
{
    public class EspecialidadBo
    {
        private EspecialidadDAO especialidadDAO;

        public EspecialidadBo()
        {
            this.especialidadDAO = new EspecialidadDAOImpl();
        }

        public int insertar(int idEspecialidad, string nombre_especialidad, string faculdad, string descripcion_nivel)
        {
            Nivel nivel = (Nivel)Enum.Parse(typeof(Nivel), descripcion_nivel);
            Especialidad especialidad = new Especialidad(idEspecialidad, nombre_especialidad, faculdad, nivel);
            return this.especialidadDAO.insertar(especialidad); 
        }

        public int modificar(int idEspecialidad, string nombre_especialidad, string faculdad, string descripcion_nivel)
        {
            Nivel nivel = (Nivel)Enum.Parse(typeof(Nivel), descripcion_nivel);
            Especialidad especialidad = new Especialidad(idEspecialidad, nombre_especialidad, faculdad, nivel);
            return this.especialidadDAO.modificar(especialidad);
        }

        public int eliminar(int idEspecialidad)
        {
            Especialidad especialidad = new Especialidad();
            especialidad.IdEspecialidad = idEspecialidad;
            return this.especialidadDAO.eliminar(especialidad);
        }

        public BindingList<Object> listarTodos()
        {
            return this.especialidadDAO.listarTodos();
        }

        public Especialidad obtenerPorId(int idEspecialidad)
        {
            return this.especialidadDAO.obtenerPorId(idEspecialidad);
        }
    }
}

File: 2024-2/0682/unidad07.f - Programación distribuida - reportes/ejemplo completo/SoftPub_Back-end/ServidorRMI/SoftPubPersonaBO/src/pe/edu/pucp/softpub/persona/bo/EspecialidadBO.java
package pe.edu.pucp.softpub.persona.bo;

import java.util.ArrayList;
import pe.edu.pucp.softpub.persona.dao.EspecialidadDAO;
import pe.edu.pucp.softpub.persona.daoImpl.EspecialidadDAOImpl;
import pe.edu.pucp.softpub.persona.model.Especialidad;
import pe.edu.pucp.softpub.persona.model.Nivel;

public class EspecialidadBO {

    private final EspecialidadDAO especialidadDAO;

    public EspecialidadBO() {
        this.especialidadDAO = new EspecialidadDAOImpl();
    }

    public Integer insertar(Integer idEspecialidad, String nombre_especialidad, String facultad, Nivel nivel) {
        Especialidad especialidad = new Especialidad(idEspecialidad, nombre_especialidad, facultad, nivel);
        return especialidadDAO.insertar(especialidad);
    }
    
    public Integer modificar(Integer idEspecialidad, String nombre_especialidad, String facultad, Nivel nivel) {
        Especialidad especialidad = new Especialidad(idEspecialidad, nombre_especialidad, facultad, nivel);
        return especialidadDAO.modificar(especialidad);                
    }
    
    public Integer eliminar(Integer idEspecialidad){
        Especialidad especialidad = new Especialidad();
        especialidad.setIdEspecialidad(idEspecialidad);
        return especialidadDAO.eliminar(especialidad);
    }
    
    public ArrayList<Especialidad> listarTodos(){
        return especialidadDAO.listarTodos();
    }
    
    public Especialidad obtenerPorId(Integer idEspecialidad){
        return especialidadDAO.obtenerPorId(idEspecialidad);
    }
}
